sympy/functions/elementary/miscellaneous.py

```diff
@@ -606,24 +605,24 @@ def _is_connected(cls, x, y):
         """
         Check if x and y are connected somehow.
         """
-        def hit(v, t, f):
-            if not v.is_Relational:
-                return t if v else f
         for i in range(2):
             if x == y:
                 return True
-            r = hit(x >= y, Max, Min)
-            if r is not None:
-                return r
-            r = hit(y <= x, Max, Min)
-            if r is not None:
-                return r
-            r = hit(x <= y, Min, Max)
-            if r is not None:
-                return r
-            r = hit(y >= x, Min, Max)
-            if r is not None:
-                return r
+            t, f = Max, Min
+            for op in range(2):
+                for j in range(2):
+                    try:
+                        if op == 0:
```

Comment:
I guess this can be written as `if op:` and swap the `v`-assignments, but OK as this as well.